packages-ts/starknet-gauntlet/src/wallet/ledgerWallet.ts

```diff
@@ -0,0 +1,117 @@
+import { IStarknetWallet } from './'
+import { SignerInterface, encode } from 'starknet'
+import { Abi, Invocation, InvocationsSignerDetails } from 'starknet/types'
+import { TypedData, getMessageHash } from 'starknet/utils/typedData'
+import { fromCallsToExecuteCalldataWithNonce } from 'starknet/utils/transaction'
+import { calculcateTransactionHash, getSelectorFromName } from 'starknet/utils/hash'
+import Stark, { LedgerError } from '@ledgerhq/hw-app-starknet'
+
+// EIP-2645 path
+//  2645 - EIP number
+//  579218131 - layer - 31 lowest bits of sha256("starkex")
+//  894929996 - application - 31 lowest bits of sha256("chainlink")
+const DEFAULT_PATH = "m/2645'/579218131'/894929996'/0'"
+const PATH_REGEX = /^\s*m\s*\/\s*2645\s*\'\s*\/\s*579218131\s*\'\s*\/\s*(\d+)\s*\'\s*\/\s*(\d+)\s*\'$/
+
+class LedgerSigner implements SignerInterface {
+  ledgerConnector: Stark
+  path: string
+  publicKey: string
+
+  private constructor(lc: Stark, path: string, publicKey: string) {
+    this.ledgerConnector = lc
```

Comment:
Let's just call this a client:

```suggestion
  private constructor(client: Stark, path: string, publicKey: string) {
    this.client = client
```